kmip/core/attributes.py

```diff
@@ -354,6 +379,21 @@ def __validate(self):
                 msg += "; expected {0}, received {1}".format(
                     self.KeyRoleType, self.key_role_type)
                 raise TypeError(msg)
+        if self.digital_signature_algorithm is not None:
+            if not isinstance(self.digital_signature_algorithm,
+                              self.DigitalSignatureAlgorithm):
```

Comment:
I do not quite understand.  DigitalSignatureAlgorithm is a class defined in CryptographicParameters just like KeyRoleType and etc.... What's the problem for using "self.DigitalSignatureAlgorithm"?